cl/scrapers/tasks.py

```diff
@@ -125,21 +142,30 @@ def make_pdftotext_process(path):
     )
 
 
-def extract_from_pdf(path, opinion, do_ocr=False):
+def extract_from_pdf(
+    path: str, opinion: Opinion, ocr_available: bool = False
+) -> Tuple[str, bytes]:
     """Extract text from pdfs.
 
-    Here, we use pdftotext. If that fails, try to use tesseract under the
-    assumption it's an image-based PDF. Once that is complete, we check for the
-    letter e in our content. If it's not there, we try to fix the mojibake
-    that ca9 sometimes creates.
+    Start with pdftotext.  If we we enabled OCR - and the the content is empty
+    or the PDF contains images use tesseract.  This pattern occurs because
+    PDFs can be images, text-based and a mix of the two.  We check for images
+    to avoid not OCRing mix type PDFs.
+
+    If a text-based PDF we fix corrupt PDFs from ca9.
     """
     process = make_pdftotext_process(path)
     content, err = process.communicate()
     content = content.decode()
-    if content.strip() == "" and do_ocr:
-        success, content = extract_by_ocr(path)
+
+    has_images = check_pdf_for_images(path)
+    if (content.strip() == "" or has_images) and ocr_available:
```

Comment:
Ya know, I didn't like the change to `ocr_available` at first, but I'm glad I didn't ask for something else. It's grown on me. One thing I'd change though is that we should just have an `ocr_needed` function that wraps up all the logic, so that the line can become:

    if ocr_needed and ocr_avaiable:

Inside ocr_needed, we can check the length of the content and if there are images, then return a bool as to whether ocr is needed for the doc. That'll move all that logic out of this function, making this function a lot simpler?

What do you think?

Reply:
I think its confusing to need both.

Reply:
If you say OCR is needed, Do the OCR and use it.  Right?

Reply:
I think one function to determine if we *should* run OCR and another to run it if it's available makes sense. Otherwise you wind up with something like, `is_ocr_needed_and_available`.

If we split it up, we can also flip the logic here so that `is_available` is checked first. For example, this:

    if ocr_available and ocr_needed:

Would only do the `ocr_needed` test if ocr_avaialble is True. If ocr_available is False, we wouldn't have to run the second part of the if statement.